src/install/migration.zig

```diff
@@ -937,6 +942,45 @@ pub fn migrateNPMLockfile(this: *Lockfile, allocator: Allocator, log: *logger.Lo
     this.buffers.resolutions.items.len = (@intFromPtr(resolutions_buf.ptr) - @intFromPtr(this.buffers.resolutions.items.ptr)) / @sizeOf(Install.PackageID);
     this.buffers.dependencies.items.len = this.buffers.resolutions.items.len;
 
+    // consult the corresponding package.json file for overrides, if possible
+    var overrides_copied = copied: {
+        var package_json_dir_path = path[0..(path.len - "package-lock.json".len)];
+        const package_json_path_buf = try this.allocator.alloc(u8, path.len - "package-lock.json".len + "package.json".len);
+        defer this.allocator.free(package_json_path_buf);
+        @memcpy(package_json_path_buf[0..package_json_dir_path.len], package_json_dir_path);
+        package_json_path_buf[package_json_dir_path.len..package_json_path_buf.len][0.."package.json".len].* = "package.json".*;
+
+        var json_file = bun.openFile(package_json_path_buf, .{ .mode = .read_only }) catch break :copied false;
```

Comment:
Thank you for this, but we shouldn't read package-lock.json twice. `json` is already accessible in this function, which contains the package-lock.json file.

Reply:
This is loading the _package.json_ file, to temporarily generate a bun Lockfile from it, because the package-lock.json does not contain enough information about overrides.